### PR TITLE
IJSDK-474 Clarify definition of "IDEA jdk" in README.md

### PR DESCRIPTION
**Pull request by JohnHake**
This change attempts to clarify whether users should define the IDEA jdk based on JDK 1.6 or 1.8.

I found explaining this definition to be tricky because there are three audiences: plugin developers, internal engineers, and contributors to the IntelliJ IDEA Community Edition open source project.  I tried to reach the plugin developer audience first, and clarify which JDK should be used based on how a build will be used. 

/cc @yole

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,12 +41,13 @@ Kotlin), [enable, upgrade, or install that plugin](https://www.jetbrains.com/hel
...
-   named "**1.8**", pointing to `<JDK_18_HOME>`.
-   * If not already present, add `<JDK_18_HOME>/lib/tools.jar` [to the Classpath](https://www.jetbrains.com/help/idea/sdk.html#manage_sdks) tab for the **1.8** JSDK.
-2. Also configure a JSDK named "**IDEA jdk**" (case sensitive), pointing to `<JDK_16_HOME>`. If you don’t want to install JDK 1.6 then you may configure **IDEA jdk** to point to `<JDK_18_HOME>`. However, you must be careful to avoid using Java 8 APIs in IntelliJ IDEA Community Edition modules that use **IDEA jdk**. 
+JDK version 1.8 is required for building and developing for IntelliJ IDEA Community Edition.
+1. Using IntelliJ IDEA, [configure](https://www.jetbrains.com/help/idea/sdk.html) a JSDK named "**1.8**", pointing to `<JDK_18_HOME>`.
+   * If not already present, add `<JDK_18_HOME>/lib/tools.jar` [to the Classpath](https://www.jetbrains.com/help/idea/sdk.html#manage_sdks) tab for the **1.8** JSDK.  
+2. Configure a JSDK named "**IDEA jdk**" (case sensitive) pointing to `<JDK_18_HOME>`. This is the recommended build configuration if you are developing 
+   new plugin projects using the IntelliJ Platform SDK.
```
Comment (chashnikov) [inline README.md]: If you're developing a plugin project there is no need to configure 'IDEA jdk' or '1.8' JDK at all. You either need to use gradle-intellij-plugin, or setup an 'IntelliJ Platform Plugin SDK'.

File: README.md [hunk 2]
```diff
@@ -41,12 +41,13 @@ Kotlin), [enable, upgrade, or install that plugin](https://www.jetbrains.com/hel
...
-2. Also configure a JSDK named "**IDEA jdk**" (case sensitive), pointing to `<JDK_16_HOME>`. If you don’t want to install JDK 1.6 then you may configure **IDEA jdk** to point to `<JDK_18_HOME>`. However, you must be careful to avoid using Java 8 APIs in IntelliJ IDEA Community Edition modules that use **IDEA jdk**. 
+JDK version 1.8 is required for building and developing for IntelliJ IDEA Community Edition.
+1. Using IntelliJ IDEA, [configure](https://www.jetbrains.com/help/idea/sdk.html) a JSDK named "**1.8**", pointing to `<JDK_18_HOME>`.
+   * If not already present, add `<JDK_18_HOME>/lib/tools.jar` [to the Classpath](https://www.jetbrains.com/help/idea/sdk.html#manage_sdks) tab for the **1.8** JSDK.  
+2. Configure a JSDK named "**IDEA jdk**" (case sensitive) pointing to `<JDK_18_HOME>`. This is the recommended build configuration if you are developing 
+   new plugin projects using the IntelliJ Platform SDK.  
+   However, if you are configuring a build with the intention of modifying the IntelliJ IDEA codebase, then the recommended build configuration
```
Comment (chashnikov) [inline README.md]: Most of the modules in IntelliJ IDEA sources uses Java 8 and therefore don't depend on 'IDEA jdk'. If you don't plan to edit code in modules which depend on 'IDEA jdk' there is no need to use Java 1.6 (see [internal docs](https://confluence.jetbrains.com/display/IDEA/IntelliJ+IDEA+project+structure+FAQ) for details).

**Comment by chashnikov** [comment]
>  there are three audiences: plugin developers, ...

Plugin developers don't need to setup and build intellij-community project. If they don't use gradle-intellij-plugin which downloads sources automatically they may check out intellij-community sources (see [the docs](http://www.jetbrains.org/intellij/sdk/docs/basics/getting_started/setting_up_environment.html) but they don't need to setup that project or build it).

**Comment by JohnHake** [comment]
Thanks for your feedback. I was trying to clarify the wording because it apparently confused a developer. However, this might be a case of "better" being the enemy of good.

How about I delete this pull request and backout the changes from my fork?

**Comment by chashnikov** [comment]
Ok, I'm closing this pull request.

**Pull request closed without merging**